fault/magma_utils.py

```diff
@@ -0,0 +1,6 @@
+import magma as m
+
+
+def is_recursive_type(T):
+    return issubclass(T, m.Tuple) or issubclass(T, m.Array) and \
```

Comment:
going forward let's try to stick to this for long lines: http://google.github.io/styleguide/pyguide.html#3-python-style-rules, in particular 

> Do not use backslash line continuation except for `with` statements requiring three or more context managers.

So

```python
return (issubclass(T, m.Tuple) or issubclass(T, m.Array) and
        not issubclass(T.T, m.Digital))
```